cluster/addons/dns/skydns-rc.yaml.in

```diff
@@ -19,6 +19,9 @@ spec:
         version: v11
         kubernetes.io/cluster-service: "true"
     spec:
+{% if grains['cloud'] is defined and grains['cloud'] in [ 'vsphere', 'photon-controller' ] %}
```

Comment:
I'm going to rollback the changes to this single file. @thockin Why was this not caught in CI? We have multiple DNS e2e tests. I also pinged the cluster team for review and got no response. I'm not sure what more we could have done to avoid this.

This change adds valid jinja to a jinja template. It's unreasonable that we can't make a change like this locally.